### PR TITLE
Improve: error handling

### PR DESCRIPTION
**Pull request by vadymmarkov**
⚠️ **Breaking change**

This PR aims to fix https://github.com/hyperoslo/Malibu/issues/84 by adding `Response` to error types when it's possible. It also renames a couple of `Response`'s properties.

```swift
public enum NetworkError: Error {
  case invalidRequestURL
  case missingContentType(response: Response)
  case invalidParameter
  case noDataInResponse
  case noResponseReceived
  case unacceptableStatusCode(statusCode: Int, response: Response)
  case unacceptableContentType(contentType: String, response: Response)
  case jsonArraySerializationFailed(response: Response)
  case jsonDictionarySerializationFailed(response: Response)
  case stringSerializationFailed(encoding: UInt, response: Response)
}
```

### CODE REVIEW

File: MalibuTests/Specs/Encoding/FormURLEncoderSpec.swift [hunk 1]
```diff
@@ -19,7 +18,7 @@ class FormURLEncoderSpec: QuickSpec {
           let data = string.data(using: String.Encoding.utf8,
                                  allowLossyConversion: false)
 
-          expect{ try encoder.encode(parameters: parameters) }.to(equal(data))
+          expect { try encoder.encode(parameters: parameters) }.to(equal(data))
```
Comment (onmyway133) [inline MalibuTests/Specs/Encoding/FormURLEncoderSpec.swift]: 👍

File: MalibuTests/Specs/Request/RequestStorageSpec.swift [hunk 2]
```diff
@@ -2,8 +2,7 @@
 import Quick
 import Nimble
 
-class RequestStorageSpec: QuickSpec {
-
+final class RequestStorageSpec: QuickSpec {
```
Comment (onmyway133) [inline MalibuTests/Specs/Request/RequestStorageSpec.swift]: 👍 for all the `final`

**Comment by zenangst** [comment]
If this is a breaking change, shouldn't that be a new major version? 🤔

**Comment by vadymmarkov** [comment]
@zenangst I thought about that, but it's not that breaking as you may think. It's breaking only if you actively use `Response` type and error types. It just doesn't look as a new major version for me.

**Pull request merged**